### DISCUSSION TITLE
Assign MS Teams and Azure AD Security Groups to file and folder permissions in SharePoint using Powershell. MS Teams and Azure AD Groups can be added using the UI.

### DISCUSSION THREAD

**Discussion by wil16098** [pnp/powershell#2910] [General]
One of our goals is to assign MS Teams members and members of Azure AD Security Groups to file and folder permissions in SharePoint using Powershell. MS Teams and Azure AD Groups can be added using the UI.

Currently, this does not seem to be working for me. This example using a SharePoint Group works:
`Set-PnPFolderPermission -List "$library" -Group "Files Visitors" -AddRole "Edit" -Identity "$library/$folder"`

But these examples using a MS Team and an Azure AD Group (respectively) do not:
```
Set-PnPFolderPermission -List "$library" -Group "$folder Team Members" -AddRole "Edit" -Identity "$library/$folder"
Set-PnPFolderPermission -List "$library" -Group "AADSG-$($folder)Admins" -AddRole "Edit" -Identity "$library/$folder"
```

Both the Team in question and the Azure AD Group in question exist and the variables resolve to the correct names. Both the Team and the Azure AD Group can be added and given permissions in the UI. Attempting to add via the example commands above gives an error. Example error:
```
Set-PnPFolderPermission : Group cannot be found.
At line:1 char:1
+ Set-PnPFolderPermission -List "$library" -Group "AADSG-$($folder)Admi ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Set-PnPFolderPermission], PSInvalidOperationException
    + FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.Files.SetFolderPermission
```

I take it only SharePoint Groups are allowed and MS Team and Azure AD Group permissions must be added manually? If there is a good way to accomplish this within a script, I would be pleased to have any hints :)
We have tried with various combinations of $group.ObjectID, etc. to no avail. I appreciate the responses!

_Originally posted by @wil16098 in https://github.com/pnp/powershell/discussions/2889_

**Comment by nandotech** [comment]
This works.

Pass the groupname as -User flag instead of -Group.
`Set-PnpFolderPermission -List '$library' -Identity '%library/$folder' -User '$groupName' -AddRole 'Edit'`
